pkg/gatewayserver/io/udp/firewall_test.go

```diff
@@ -148,4 +149,57 @@ func TestMemoryFirewall(t *testing.T) {
 			time.Sleep(tc.WaitAfter)
 		})
 	}
+
+	t.Run("FloodCheck", func(t *testing.T) {
+		ctx := test.Context()
+
+		for _, tc := range []struct {
+			rateLimiting bool
+			verify       func(error) bool
+		}{
+			{
+				rateLimiting: true,
+				verify:       func(err error) bool { return errors.IsResourceExhausted(err) },
+			},
+			{
+				rateLimiting: false,
+				verify:       func(err error) bool { return err == nil },
+			},
+		} {
+			t.Run(fmt.Sprintf("RateLimiting=%v", tc.rateLimiting), func(t *testing.T) {
```

Comment:
High coupling; I would prefer keeping the tests for rate limiting separate

Reply:
Will move into separate `firewall_ratelimiting_test.go`